### PR TITLE
added default realm, added password grant type

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
Restored #64 PR

### CODE REVIEW

**Comment by AndrewChubatiuk** [comment]
@mrparkers fixed all your PR comments and added 4 pipelines to test different keycloak providers configs. Gave you an access to push to my fork.

**Comment by AndrewChubatiuk** [comment]
Removed `default_realm` from provider configuration, cause it doesn't work as expected.

File: .circleci/config.yml [hunk 1]
```diff
@@ -30,3 +31,20 @@ jobs:
...
+      KEYCLOAK_CLIENT_SECRET: 884e0f95-0f42-4a63-9b1f-94274655669e
+  password:
+    <<: *test-template
+    environment:
+      GO111MODULE: "on"
+      KEYCLOAK_CLIENT_ID: admin-cli
+      KEYCLOAK_URL: http://localhost:8080
+      KEYCLOAK_REALM: master
+      KEYCLOAK_USER: keycloak
+      KEYCLOAK_PASSWORD: password
```
Comment (mrparkers) [inline .circleci/config.yml]: I like this idea.  I'll have to update the branch protection to require these new statuses set by CircleCI.

File: provider/keycloak_custom_user_federation.go [hunk 2]
```diff
@@ -25,7 +26,7 @@ func resourceKeycloakCustomUserFederation() *schema.Resource {
 			},
 			"realm_id": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
```
Comment (mrparkers) [inline provider/keycloak_custom_user_federation.go]: Since default realm was removed, this should be changed back.

**Comment by mrparkers** [comment]
After the comment is addressed, I'll merge this and update the docs to explain the new provider attributes and auth type.  Once that is done, I'll cut a release so you can start using this.

Thanks again for the contribution!

**Comment by AndrewChubatiuk** [comment]
Oh, forgot to revert this change as well

**Comment by AndrewChubatiuk** [comment]
Everything is ready now

**Comment by mrparkers** [comment]
@AndrewChubatiuk I'll try and get the docs updated in the next couple of days and cut a release for you.

**Pull request merged**